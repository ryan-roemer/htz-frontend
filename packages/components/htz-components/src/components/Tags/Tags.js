import React from 'react';
import PropTypes from 'prop-types';
import { FelaComponent, } from 'react-fela';
import { parseStyleProps, } from '@haaretz/htz-css-tools';
import { stylesPropType, } from '../../propTypes/stylesPropType';
import HtzLink from '../HtzLink/HtzLink';
import H from '../AutoLevels/H';

const propTypes = {
  /**
   * An array of tags.
   */
  tagsList: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * Tag's url.
       */
      url: PropTypes.string.isRequired,
      /**
       * Tag's name to be displayed.
       */
      contentName: PropTypes.string.isRequired,
    })
  ).isRequired,
  /**
   * A special property holding miscellaneous CSS values that
   * trumps all default values. Processed by
   * [`parseStyleProps`](https://Haaretz.github.io/htz-frontend/htz-css-tools#parsestyleprops)
   */
  miscStyles: stylesPropType,
};

const defaultProps = {
  miscStyles: null,
};

const wrapperStyle = ({ miscStyles, theme, }) => ({
  ...theme.type(-1),
  display: 'flex',
  extend: [ ...(miscStyles ? parseStyleProps(miscStyles, theme.mq, theme.type) : []), ],
});

/*
 * Returns an inline list of the article's related tags.
 */
function Tags({ tagsList, miscStyles, }) {
  return tagsList && tagsList.length > 0 ? (
    <FelaComponent
      miscStyles={miscStyles}
      rule={wrapperStyle}
      render={({ className, theme, }) => (
        <div className={className}>
          <FelaComponent
            style={{
              color: theme.color('secondary'),
              marginEnd: '1rem',
            }}
            render={({ className, }) => (
              <H className={className}>{theme.tagsElementI18n.prefix}</H>
            )}
          />
          <ul>
            {tagsList.map((tag, i) => (
              <FelaComponent
                style={{
                  display: 'inline',
                  ...(tagsList.length !== i + 1 ? { marginEnd: '1rem', } : {}),
                }}
                key={tag.contentName}
                render="li"
              >
                <HtzLink
                  href={tag.url}
                  content={(
                    <FelaComponent
                      style={{
                        color: theme.color('neutral', '-2'),
                        textDecoration: 'underline',
                      }}
                      render="span"
                    >
                      {tag.contentName}
                    </FelaComponent>
)}
                />
              </FelaComponent>
            ))}
          </ul>
        </div>
      )}
    />
  ) : null;
}

Tags.propTypes = propTypes;
Tags.defaultProps = defaultProps;

export default Tags;
