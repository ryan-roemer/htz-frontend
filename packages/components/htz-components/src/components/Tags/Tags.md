<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Tags example](#tags-example)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### Tags example

```jsx
<div style={{ direction: 'rtl'}}>
  <Tags
    tagsList={[
      {
        "url": "http://eran.themarker.com:8080/hayehudim-baim-1.2670",
        "contentName": "היהודים באים"
      },
      {
        "url": "/food/dairy-1.2429",
        "contentName": "חלבי"
      },
      {
        "url": "/news/terror-1.2603",
        "contentName": "טרור"
      }
    ]}
  />
</div>
```
