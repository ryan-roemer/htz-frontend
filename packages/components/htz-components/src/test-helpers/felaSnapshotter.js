import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
