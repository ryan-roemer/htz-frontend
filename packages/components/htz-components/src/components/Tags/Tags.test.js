import React from 'react';
import felaSnapshotter from '../../test-helpers/felaSnapshotter';
import Tags from './Tags';

it('MobileQuickRegistration ', () => {
  const snapshot = felaSnapshotter(
    <Tags
      tagsList={[
        {
          url: 'http://eran.themarker.com:8080/hayehudim-baim-1.2670',
          contentName: 'היהודים באים',
        },
        {
          url: '/food/dairy-1.2429',
          contentName: 'חלבי',
        },
        {
          url: '/news/terror-1.2603',
          contentName: 'טרור',
        },
      ]}
    />
  );
  expect(snapshot).toMatchSnapshot();
});
