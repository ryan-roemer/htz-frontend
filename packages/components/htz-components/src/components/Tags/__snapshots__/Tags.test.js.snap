// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileQuickRegistration  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e"
    >
      <h2
        className="htz-f htz-g"
      >
        תגיות:
      </h2>
      <ul>
        <li
          className="htz-h htz-g"
        >
          <a
            className={null}
            href="http://eran.themarker.com:8080/hayehudim-baim-1.2670"
            onClick={[Function]}
          >
            <span
              className="htz-i htz-j"
            >
              היהודים באים
            </span>
          </a>
        </li>
        <li
          className="htz-h htz-g"
        >
          <a
            className={null}
            href="/food/dairy-1.2429"
            onClick={[Function]}
          >
            <span
              className="htz-i htz-j"
            >
              חלבי
            </span>
          </a>
        </li>
        <li
          className="htz-h"
        >
          <a
            className={null}
            href="/news/terror-1.2603"
            onClick={[Function]}
          >
            <span
              className="htz-i htz-j"
            >
              טרור
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "styles": ".htz-e{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}.htz-f{color:#00537A}.htz-g{margin-left:1rem}.htz-h{display:inline}.htz-i{color:#505050}.htz-j{text-decoration:underline}@media (max-width: 79.9375em){.htz-a{font-size:2.3333333333333335rem}.htz-b{line-height:1.2857142857142858em}}@media (min-width: 80em){.htz-c{font-size:2.2857142857142856rem}.htz-d{line-height:1.3125em}}",
}
`;
